File: Samples/code/modpow.cpp
int modPow(int a, int b, int m)
{
    int res = 1;
    for (; b > 0; b >>= 1)
    {
        if (b & 1)
            res = (long long)res * a % m;
        a = (long long)a * a % m;
    }
    return res;
}

File: Samples/code/modpow.py
def modPow(a, b, m):
    res = 1

    while(b > 0):
        if (b & 1 == 1):
            res = res * a % m
        a = a * a % m

        b >>= 1
    
    return res

n = int(input())
v = modPow(2, n, 86400) - 1

print("%02d:%02d:%02d" % (v / 3600, (v / 60) % 60, v % 60))